### ISSUE TITLE
Accessibility: Mini Cart block button missing accessible name

### ISSUE THREAD

**Issue by webmandesign** [woothemes/woocommerce#42289]
## Describe the bug

Checking my block theme powered WooCommerce website for accessibility issues with https://wave.webaim.org/ (and also Chrome's [Lighthouse accessibility audit](https://developer.chrome.com/docs/lighthouse/overview/)) I get an error of Mini Cart button containing no accessible name.

The issue only happens when the page is loaded as the block produces a button without any `aria-label` attribute.

When I hover over the Mini Cart button, its JavaScript kicks in and adds `aria-label` attribute dynamically.

However, the accessibility evaluation tools mentioned above work with JavaScript disabled, so you can reproduce the issue when you disable JavaScript in your browser too.

## To reproduce

1. Add a Mini Cart block (`woocommerce/mini-cart`) into site header, for example.
2. Disable JavaScript in your browser and inspect the Mini Cart button HTML using browser code inspector to see there is no text or `aria-label` set for the button.

## Expected behavior

Every button should have a label, text.

## Screenshots

![wc-mini-cart-a11y-issue](https://github.com/woocommerce/woocommerce-blocks/assets/2456735/d8e4639b-aa82-4cd3-9651-d289575d4cd1)

### WordPress (please complete the following information):

-   WordPress version: 6.2.2
-   WooCommerce version: 7.9.0

**Comment by nefeline** [comment]
Hi @webmandesign ! Thanks for reaching out to us and reporting this issue: I'm adding this to our backlog, more updates will be shared here as soon as we start working on it.

**Comment by webmandesign** [comment]
Meanwhile I've fixed this in my theme using code similar to:

```php
add_filter( 'render_block', function( $block_content, $block ) {

	if ( 'woocommerce/mini-cart' === $block['blockName'] ) {
		$html = new WP_HTML_Tag_Processor( $block_content );

		$html->next_tag( array( 'class_name' => 'wc-block-mini-cart__button' ) );
		$html->set_attribute( 'aria-label', __( 'Shopping cart', 'woocommerce' ) );

		$block_content = $html->get_updated_html();
	}

	return $block_content;

}, 10, 2 );
```

**Comment by amberhinds** [comment]
I can confirm that is is still an issue. It would be geat to see a fix.

**Comment by jamesckemp** [comment]
@amberhinds It looks like this will be on its way to core soon: https://github.com/woocommerce/woocommerce-blocks/pull/9672

**Comment by webmandesign** [comment]
Hi @jamesckemp,

Thank you for pointing this out. From what I can see, the PR you have shared is not resolving the issue described here. And it seems the code from the PR is already applied in current WooCommerce plugin.

My issue is not related to JavaScript. Basically, before JavaScript kicks in, there is no label on the Mini Cart. This throws accessibility error with WAVE, for example. This is the issue I'm reporting here.

**Comment by jamesckemp** [comment]
Ah thank you for the clarification @webmandesign. Let me dig into that.

**Comment by alexflorisca** [comment]
@webmandesign can you share your error here? I don't get any Lighthouse errors for this. And while [Lighthouse does turn off javascript for some checks ](https://developer.chrome.com/docs/lighthouse/pwa/without-javascript#:~:text=Lighthouse%20disables%20JavaScript%20on%20the,Installable%2C%20and%20PWA%20optimized).), I think it's enabled for most other checks. Alas, I think this is probably good practice so I've done a PR for it anyway, but I would love to see the error your getting (and maybe some more steps/screenshots about how to replicate it)

**Comment by webmandesign** [comment]
Hi @alexflorisca 

Please note that this issue is almost a year old and I don't remember if Lighthouse test worked with JS disabled. 
Nowadays it does not - I always get some error in Lighthouse when I disable JS in browser.

Anyway, you can still test with WAVE: https://wave.webaim.org/report#/https://themedemos.webmandesign.eu/zooey/wc-a11y-issue-test/

![2482026342](https://github.com/woocommerce/woocommerce/assets/2456735/29bf42fd-91ae-4cef-9e76-f4ec4d3cb734)

---

**PS**: Please, let me know once you are finished with the test so I can remove that test page from my servers. Thanks

**Comment by alexflorisca** [comment]
@webmandesign The PR is linked above that will solve this issue. Issue will be closed when the PR gets merged so you should get notified. It will likely land in WC 9.1 so you may have to wait a month or so before it's live
